From 8eee678f40cbb5d5e0cef0940b6502452a60868c Mon Sep 17 00:00:00 2001
From: "Radu Pirea (NXP OSS)" <radu-nicolae.pirea@oss.nxp.com>
Date: Wed, 11 May 2022 15:10:18 +0300
Subject: [PATCH 23/64] s32-cc: fix linux overlap with device tree

Fix linux overlap with device tree if the linux image is bigger than
20MB.

Issue: ALB-8871

Upstream-Status: Pending 

Signed-off-by: Radu Pirea (NXP OSS) <radu-nicolae.pirea@oss.nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 include/configs/s32-cc.h | 18 ++++++++++++------
 1 file changed, 12 insertions(+), 6 deletions(-)

diff --git a/include/configs/s32-cc.h b/include/configs/s32-cc.h
index dbcef18f8b..0afd916075 100644
--- a/include/configs/s32-cc.h
+++ b/include/configs/s32-cc.h
@@ -27,16 +27,22 @@
 #define S32CC_SRAM_BASE			0x34000000
 
 /**
+ *
+ * Before changing the device tree offset or size, please read
+ * https://docs.kernel.org/arm64/booting.html#setup-the-device-tree
+ *
  * DDR images layout
  *
- * Name		Size	Address
+ * Name				Size	Address
  *
- * Image	20M	CONFIG_SYS_LOAD_ADDR
- * Linux DTB	64K	CONFIG_SYS_LOAD_ADDR + 20M
- * Ramdisk	-	CONFIG_SYS_LOAD_ADDR + 20M + 64K
+ * Image			48M	CONFIG_SYS_LOAD_ADDR
+ * Linux DTB			2M	CONFIG_SYS_LOAD_ADDR + 48M
+ * Reserved memory regions	206	CONFIG_SYS_LOAD_ADDR + 50M
+ * Ramdisk			-	CONFIG_SYS_LOAD_ADDR + 256M
  */
-#define S32CC_FDT_ADDR			0x81400000
-#define S32CC_RAMDISK_ADDR		0x81410000
+#define S32CC_FDT_ADDR			0x83000000
+#define S32CC_RAMDISK_ADDR		0x90000000
+
 /* Disable Ramdisk & FDT relocation*/
 #define S32CC_INITRD_HIGH_ADDR		0xffffffffffffffff
 #define S32CC_FDT_HIGH_ADDR		0xffffffffffffffff
-- 
2.17.1

